server/src/server.ts

```diff
@@ -388,7 +388,13 @@ export default class BashServer {
 
     if (word) {
       // Filter to only return suffixes of the current word
-      return allCompletions.filter(item => item.label.startsWith(word))
+      const validCompletions = allCompletions.filter(item => item.label.startsWith(word))
```

Comment:
Thanks for trying fixing this and reporting the error. 

Actually, I cannot see why we ever want to return any completions when the word is `{`. 🤔

So I suggest we simply do an early exit after the `if (word && word.startsWith('#')) {` block with:

```typescript
if (word === '{') {
  // We should not complete when it is not prefixed by a $. This case needs to be
  // as { is a completionProvider triggerCharacter.
  return []
}
```

It would also be great to document that we don't complete on `{`. It could be line `4` in the issue101.sh file.

Reply:
:+1: I redid the commit above, and I added a test case. ready for review.